tmt/convert.py

```diff
@@ -46,6 +46,116 @@ def repr_str(dumper, data):
+    """
+    Reads metadata of manual test cases from Nitrate
+    """
+    try:
+        import nitrate
+    except ImportError:
+        raise ConvertError('Install nitrate module to import metadata.')
+    # Turns off nitrate caching
+    nitrate.set_cache_level(0)
+
+    try:
+        tree = fmf.Tree(os.getcwd())
+    except fmf.utils.RootError:
+        raise ConvertError("Initialize metadata tree using 'tmt init'")
+
+    try:
+        if plan_id:
+            all_cases = nitrate.TestPlan(int(plan_id)).testcases
+            case_ids = [case.id for case in all_cases if case.manual]
+        else:
+            case_ids = [int(case_id)]
+    except ValueError:
+        raise ConvertError('Value of plan/case option must be an integer')
+
+    # Create directory to store manual tests in
+    old_cwd = os.getcwd()
+    os.chdir(tree.root)
+    try:
+        os.mkdir('Manual')
+    except FileExistsError:
+        pass
+
+    os.chdir('Manual')
+
+    for case_id in case_ids:
+        testcase = nitrate.TestCase(case_id)
+        if testcase.status.name != 'CONFIRMED' and not disabled:
+            continue
+
+        # Filename sanitization
+        dir_name = testcase.summary.replace(' ', '_')
+        dir_name = dir_name.replace('/', '_')
+        try:
+            os.mkdir(dir_name)
+        except FileExistsError:
+            pass
+
+        os.chdir(dir_name)
+        echo("Importing the '{0}' test case.".format(dir_name))
+
+        # Test case data
+        md_content = {}
+        md_content['Setup'] = testcase.setup
+        md_content['Actions'] = testcase.action
+        md_content['Breakdown'] = testcase.breakdown
+        md_content['Expected Results'] = testcase.effect
+
+        # Test case metadata
+        data = {}
+        data['manual'] = True
+        data['test'] = 'full.md'
+
+        # Relevancy
+        field = tmt.utils.StructuredField(testcase.notes)
+        try:
+            relevancy = field.get('relevancy')
+            if relevancy:
+                data['relevancy'] = relevancy
+                echo(style('relevancy:', fg='green'))
+                echo(data['relevancy'].rstrip('\n'))
+        except tmt.utils.StructuredFieldError:
+            pass
+        # Contact
+        if testcase.tester:
+            data['contact'] = '{} <{}>'.format(
+                testcase.tester.name, testcase.tester.email)
+            echo(style('contact: ', fg='green') + data['contact'])
+        # Test identifier
+        data['extra-nitrate'] = testcase.identifier
+        echo(style('extra-nitrate: ', fg='green') + data['extra-nitrate'])
+        # Status
+        data['enabled'] = testcase.status.name == "CONFIRMED"
+        echo(style('enabled: ', fg='green') + str(data['enabled']))
+
+        write_markdown(os.getcwd() + '/full.md', md_content)
+        write(os.getcwd() + '/main.fmf', data)
+        os.chdir('..')
+
+    os.chdir(old_cwd)
+
+
+def write_markdown(path, content):
```

Comment:
Would it make sense to use a [python markdown module](https://pypi.org/project/Markdown/) for handling Markdown? If/when we are able to export manual cases I guess we will need it anyway.

Reply:
From what I could find out the markdown module it is only/mostly used to markdown -> html conversion rather than plaintext -> markdown or even html -> markdown which would also be  useful.
If you know how to use it better please let me know.

Reply:
I tried to import an example test and the result looks like this:

    # Setup
    <p>Do this and that to setup the environment.</p>

    # Actions
    <p>Step one.</p>
    <p>Step two.</p>
    <p>Step three.</p>

    # Breakdown
    <p>This is a breakdown.</p>

    # Expected Results
    <p>Expect one.</p>
    <p>Expect two</p>
    <p>Expect three.</p>

I guess the `markdown` module could/would take care of handling the html code and convert paragraph tags into new lines?

Reply:
As I said in previous comment I do not believe this is how `markdown` module works. As far as I know it is intended for converting markdown->html not the other way around.

I found a way how to *sort of* convert html to md however it uses non-standart library (html2text) and the results are always the best.

Reply:
I see, thanks for clarification. I think it adding `python3-html2text` to the dependencies should be OK. It's just a small package and it does not bring many other deps. It would be good to prevent having many `<p>` and other html tags in the imported content, even if the conversion is not always ideal.

Reply:
Looks good to me after last changes. Just rebase and squash.